From linux-ext4-owner@vger.kernel.org Mon Feb 18 16:05:20 2008
Return-Path: <linux-ext4-owner@vger.kernel.org>
Received: from imap.linux.ibm.com ([unix socket]) by imap.linux.ibm.com
	(Cyrus v2.3.7-Invoca-RPM-2.3.7-7) with LMTPA; Mon, 18 Feb 2008 16:05:20
	-0500
X-Sieve: CMU Sieve 2.3
Received: by imap.linux.ibm.com (Postfix, from userid 101) id 544C61910236;
	Mon, 18 Feb 2008 16:05:20 -0500 (EST)
X-Spam-TestScore: TW_MK=0.077
X-Spam-TokenSummary: Bayes not run.
X-Spam-Checker-Version: SpamAssassin 3.1.7 (2006-10-05) on
	imap.linux.ibm.com
X-Spam-Level: 
X-Spam-Status: No, score=0.1 required=5.0 tests=TW_MK autolearn=disabled 
	version=3.1.7
X-Spam-Relay-Country: US ** US US US ** US US US US ** ** **
Received: from smtp.linux.ibm.com (smtp.linux.ibm.com [9.26.4.197]) by
	imap.linux.ibm.com (Postfix) with ESMTP id 42736191021E for
	<cmm@imap.linux.ibm.com>; Mon, 18 Feb 2008 16:05:16 -0500 (EST)
Received: from localhost (localhost.localdomain [127.0.0.1]) by
	smtp.linux.ibm.com (Postfix) with ESMTP id E4680C040 for
	<cmm@linux.ibm.com>; Mon, 18 Feb 2008 16:05:15 -0500 (EST)
X-Virus-Scanned: amavisd-new at linux.ibm.com
Received: from BLDVMB.POK.IBM.COM (bldvmb.pok.ibm.com [9.57.6.113]) by
	smtp.linux.ibm.com (Postfix) with ESMTP id 95A79C078 for
	<cmm@linux.ibm.com>; Mon, 18 Feb 2008 16:05:15 -0500 (EST)
Received:  by BLDVMB.POK.IBM.COM (IBM VM SMTP Level 440) via spool with
	SMTP id 4901 ; Mon, 18 Feb 2008 14:06:16 MST
Received: by bldvmb.vnet.ibm.com (xagent2 5.9.5) via xagsmtp2 with spool id
	3427 for cmm@linux.vnet.ibm.com; Mon, 18 Feb 2008 14:06:16 -0700 (MST)
Received: from d03relay03.boulder.ibm.com [9.17.195.228] by
	BLDVMB.POK.IBM.COM (IBM VM SMTP Level 440) via TCP with ESMTP ; Mon, 18 Feb
	2008 14:06:16 MST
Received: from d03av05.boulder.ibm.com (d03av05.boulder.ibm.com
	[9.17.195.85]) by d03relay03.boulder.ibm.com (8.13.8/8.13.8/NCO v8.7) with
	ESMTP id m1IL5DuN137190; Mon, 18 Feb 2008 14:05:13 -0700
Received: from d03av05.boulder.ibm.com (loopback [127.0.0.1]) by
	d03av05.boulder.ibm.com (8.13.1/8.13.3) with ESMTP id m1IL5Duj027840; Mon,
	18 Feb 2008 14:05:13 -0700
Received: from d03as04.boulder.ibm.com (d03as04 [9.17.195.245]) by
	d03av05.boulder.ibm.com (8.13.1/8.12.11) with ESMTP id m1IL5DUn027837; Mon,
	18 Feb 2008 14:05:13 -0700
Received: from e34.co.us.ibm.com ([9.17.249.44]) by d03as04.boulder.ibm.com
	(8.12.11.20060308/8.12.11) with ESMTP id m1IL5BgK006157
	(version=TLSv1/SSLv3 cipher=DHE-RSA-AES256-SHA bits=256 verify=OK); Mon, 18
	Feb 2008 14:05:12 -0700
Received: from vger.kernel.org (vger.kernel.org [209.132.176.167]) by
	e34.co.us.ibm.com (8.13.8/8.13.8) with ESMTP id m1IL52Iq003127; Mon, 18 Feb
	2008 16:05:02 -0500
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand id
	S1753045AbYBRVAg (ORCPT <rfc822;jrs@us.ibm.com> + 2 others); Mon, 18 Feb
	2008 16:00:36 -0500
Received: (majordomo@vger.kernel.org) by vger.kernel.org id
	S1753575AbYBRVAg (ORCPT <rfc822;linux-ext4-outgoing>); Mon, 18 Feb 2008
	16:00:36 -0500
Received: from mx1.redhat.com ([66.187.233.31]:38086 "EHLO mx1.redhat.com"
	rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP id S1753045AbYBRVAf
	(ORCPT <rfc822;linux-ext4@vger.kernel.org>); Mon, 18 Feb 2008 16:00:35 -0500
Received: from int-mx1.corp.redhat.com (int-mx1.corp.redhat.com
	[172.16.52.254]) by mx1.redhat.com (8.13.8/8.13.8) with ESMTP id
	m1IL0ZKX010235 for <linux-ext4@vger.kernel.org>; Mon, 18 Feb 2008 16:00:35
	-0500
Received: from lacrosse.corp.redhat.com (lacrosse.corp.redhat.com
	[172.16.52.154]) by int-mx1.corp.redhat.com (8.13.1/8.13.1) with ESMTP id
	m1IL0Y2b011349 for <linux-ext4@vger.kernel.org>; Mon, 18 Feb 2008 16:00:34
	-0500
Received: from neon.msp.redhat.com (neon.msp.redhat.com [10.15.80.10]) by
	lacrosse.corp.redhat.com (8.12.11.20060308/8.11.6) with ESMTP id
	m1IL0YsE020300 for <linux-ext4@vger.kernel.org>; Mon, 18 Feb 2008 16:00:34
	-0500
Message-ID: <47B9F1F1.30604@redhat.com>
Date: 	Mon, 18 Feb 2008 15:00:33 -0600
From: Eric Sandeen <sandeen@redhat.com>
User-Agent: Thunderbird 2.0.0.9 (X11/20071115)
MIME-Version: 1.0
To: ext4 development <linux-ext4@vger.kernel.org>
Subject: [PATCH] clear extents flag on inodes created in ext4_mknod
Content-Type: text/plain; charset=ISO-8859-1
X-Scanned-By: MIMEDefang 2.58 on 172.16.52.254
Sender: linux-ext4-owner@vger.kernel.org
Precedence: bulk
List-ID: <linux-ext4.vger.kernel.org>
X-Mailing-List: 	linux-ext4@vger.kernel.org
X-Xagent-From: sandeen@redhat.com
X-Xagent-To: cmm@linux.vnet.ibm.com
X-Xagent-Gateway: bldvmb.vnet.ibm.com (XAGENTU at BLDVMB)
X-Evolution-Source: imap://cmm@imap.linux.ibm.com/
Content-Transfer-Encoding: 8bit

e2fsck doesn't expect to find char, block, fifo, or socket
files with the extent flag set, so clear that in ext4_mknod.

Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Index: linux-2.6.24/fs/ext4/namei.c
===================================================================
--- linux-2.6.24.orig/fs/ext4/namei.c
+++ linux-2.6.24/fs/ext4/namei.c
@@ -1766,6 +1766,7 @@ retry:
 #ifdef CONFIG_EXT4DEV_FS_XATTR
 		inode->i_op = &ext4_special_inode_operations;
 #endif
+		EXT4_I(inode)->i_flags &= ~EXT4_EXTENTS_FL;
 		err = ext4_add_nondir(handle, dentry, inode);
 	}
 	ext4_journal_stop(handle);


-
To unsubscribe from this list: send the line "unsubscribe linux-ext4" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html
